lib/components/AlbumScreen/download_dialog.dart

```diff
@@ -1,114 +1,214 @@
+import '../global_snackbar.dart';
 
 class DownloadDialog extends StatefulWidget {
-  const DownloadDialog({
-    Key? key,
-    required this.parents,
-    required this.items,
+  const DownloadDialog._build({
+    super.key,
+    required this.item,
     required this.viewId,
-  })  : assert(parents.length == items.length),
-        super(key: key);
+    required this.needDirectory,
+    required this.needTranscode,
+    required this.children,
+  });
 
-  final List<BaseItemDto> parents;
-  final List<List<BaseItemDto>> items;
+  final DownloadStub item;
   final String viewId;
+  final bool needDirectory;
+  final bool needTranscode;
+  final List<BaseItemDto>? children;
 
   @override
   State<DownloadDialog> createState() => _DownloadDialogState();
+
+  static Future<void> show(
+      BuildContext context, DownloadStub item, String? viewId) async {
+    if (viewId == null) {
+      final finampUserHelper = GetIt.instance<FinampUserHelper>();
+      viewId = finampUserHelper.currentUser!.currentViewId;
+    }
+    bool needTranscode =
+        FinampSettingsHelper.finampSettings.shouldTranscodeDownloads ==
+            TranscodeDownloadsSetting.ask;
+    bool needDownload = FinampSettingsHelper
+            .finampSettings.downloadLocationsMap.values
+            .where((element) =>
+                element.baseDirectory != DownloadLocationType.internalDocuments)
+            .length !=
+        1;
+    if (!needTranscode && !needDownload) {
+      final isarDownloads = GetIt.instance<IsarDownloads>();
+      var profile = FinampSettingsHelper
+                  .finampSettings.shouldTranscodeDownloads ==
+              TranscodeDownloadsSetting.always
+          ? FinampSettingsHelper.finampSettings.downloadTranscodingProfile
+          : DownloadProfile(transcodeCodec: FinampTranscodingCodec.original);
+      profile.downloadLocationId =
+          FinampSettingsHelper.finampSettings.internalSongDir.id;
+      unawaited(isarDownloads
+          .addDownload(stub: item, viewId: viewId!, transcodeProfile: profile)
+          .then((value) => GlobalSnackbar.message(
+              (scaffold) => AppLocalizations.of(scaffold)!.downloadsAdded)));
+    } else {
+      JellyfinApiHelper jellyfinApiHelper = GetIt.instance<JellyfinApiHelper>();
+      List<BaseItemDto>? children;
+      if (item.baseItemType == BaseItemDtoType.album ||
+          item.baseItemType == BaseItemDtoType.playlist) {
+        children = await jellyfinApiHelper.getItems(
+            parentItem: item.baseItem!,
+            includeItemTypes: "Audio",
+            fields: "${jellyfinApiHelper.defaultFields},MediaSources");
+      }
+      if (!context.mounted) return;
+      await showDialog(
+        context: context,
+        builder: (context) => DownloadDialog._build(
+            item: item,
+            viewId: viewId!,
+            needDirectory: needDownload,
+            needTranscode: needTranscode,
+            children: children),
+      );
```

Comment:
Because the show function doesn't actually create a dialog if there is only one download and transcoding is not set to ask.  The download button calls show() when pressed, which then either shows the dialog or decides it isn't needed and starts the download immediatly.

Reply:
I see. Would you mind adding a few comments somewhere that explain this? Or maybe I just didn't see them...